pkg/validation/validation.go

```diff
@@ -0,0 +1,83 @@
+package validation
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// https://github.com/xeipuuv/gojsonschema#formats
+type ValidFormat struct{}
+
+// IsFormat always returns true and meets the
+// gojsonschema.FormatChecker interface
+func (f ValidFormat) IsFormat(input interface{}) bool {
+	return true
+}
+
+// Based on https://stackoverflow.com/questions/40737122/convert-yaml-to-json-without-struct-golang
+// We unmarshal yaml into a value of type interface{},
+// go through the result recursively, and convert each encountered
+// map[interface{}]interface{} to a map[string]interface{} value
+// required to marshall to JSON.
+// Reference: https://github.com/garethr/kubeval/blob/master/kubeval/utils.go#L8
+func convertToStringKeys(i interface{}) interface{} {
+	switch x := i.(type) {
+	case map[interface{}]interface{}:
+		m2 := map[string]interface{}{}
+		for k, v := range x {
+			m2[k.(string)] = convertToStringKeys(v)
+		}
+		return m2
+	case []interface{}:
+		for i, v := range x {
+			x[i] = convertToStringKeys(v)
+		}
+	}
+	return i
+}
+
+// validate function will validate input kedgefile against JSON schema provided in pkg/validation
+func Validate(p []byte) {
+	var speco interface{}
+	err := yaml.Unmarshal(p, &speco)
```

Comment:
If we can't avoid this then we need explanation why we are doing it like this

Reply:
sure